BrowserKit/Sources/WebEngine/WKWebview/WebServer/WKWebServerUtil.swift

```diff
@@ -14,14 +14,19 @@ class DefaultWKWebServerUtil: WKWebServerUtil {
     private var webServer: WKEngineWebServerProtocol
     private let logger: Logger
 
-    init(webServer: WKEngineWebServerProtocol = WKEngineWebServer.shared,
+    init(webServer: WKEngineWebServerProtocol,
          readerModeHander: WKReaderModeHandlersProtocol = WKReaderModeHandlers(),
          logger: Logger = DefaultLogger.shared) {
         self.webServer = webServer
         self.readerModeHander = readerModeHander
         self.logger = logger
     }
 
+    static func make() async -> DefaultWKWebServerUtil {
+        let webServer = await MainActor.run { WKEngineWebServer.shared }
```

Comment:
`WKEngineWebServer.shared` also needs to be called on the main thread. It's creating a `GCDWebserver` which crashes if its not on the main thread